sys/include/net/lora.h

```diff
@@ -38,66 +40,164 @@ extern "C" {
 #endif
 
-/** @brief Set fixed payload length on */
-#ifndef LORA_FIX_LENGTH_PAYLOAD_ON_DEFAULT
-#define LORA_FIX_LENGTH_PAYLOAD_ON_DEFAULT     (false)
+#ifndef CONFIG_LORA_SF_DEFAULT
+#define CONFIG_LORA_SF_DEFAULT                      (LORA_SF7)
 #endif
 
-/** @brief Set inverted IQ on */
-#ifndef LORA_IQ_INVERTED_DEFAULT
-#define LORA_IQ_INVERTED_DEFAULT               (false)
+/** @brief Set Coding Rate (CR)
+ *
+ * Configure Coding Rate (CR). CR denotes the implementation of forward error
+ * correction (FEC). This may be done by encoding 4-bit data with redundancies
+ * into 5-bit, 6-bit, 7-bit, or 8-bit. Coding Rate (CR) value need to be
+ * adjusted according to conditions of the channel used for data transmission.
+ * If there are too many interferences in the channel, then it’s recommended to
+ * increase the value of CR. However, the rise in CR value will also increase
+ * the duration for the transmission. Refer to country specific air time usage
+ * regulations before varying the CR. To calculate air time refer
+ * https://www.loratools.nl/#/airtime .
+*/
+#if IS_ACTIVE(CONFIG_LORA_CR_DEFAULT_CR_4_5)
+#define CONFIG_LORA_CR_DEFAULT                      (LORA_CR_4_5)
+#elif IS_ACTIVE(CONFIG_LORA_CR_DEFAULT_CR_4_6)
+#define CONFIG_LORA_CR_DEFAULT                      (LORA_CR_4_6)
+#elif IS_ACTIVE(CONFIG_LORA_CR_DEFAULT_CR_4_7)
+#define CONFIG_LORA_CR_DEFAULT                      (LORA_CR_4_7)
+#elif IS_ACTIVE(CONFIG_LORA_CR_DEFAULT_CR_4_8)
+#define CONFIG_LORA_CR_DEFAULT                      (LORA_CR_4_8)
 #endif
 
-/** @brief Frequency hopping on */
-#ifndef LORA_FREQUENCY_HOPPING_DEFAULT
-#define LORA_FREQUENCY_HOPPING_DEFAULT         (false)
+#ifndef CONFIG_LORA_CR_DEFAULT
+#define CONFIG_LORA_CR_DEFAULT                      (LORA_CR_4_5)
 #endif
 
-/** @brief Frequency hopping period */
-#ifndef LORA_FREQUENCY_HOPPING_PERIOD_DEFAULT
-#define LORA_FREQUENCY_HOPPING_PERIOD_DEFAULT  (0U)
+/** @brief Set this to 1 to enable inverted I/Q mode
+ *
+ * Enable this to invert the IQ signals used in RF modulation circuit. For more
+ * information on I/Q modulation technique visit http://www.ni.com/tutorial/4805/en/
+*/
+#ifdef DOXYGEN
+#define CONFIG_LORA_IQ_INVERTED_DEFAULT
 #endif
 
-/** @brief Set fixed header length mode (implicit header) */
-#ifndef LORA_FIXED_HEADER_LEN_MODE_DEFAULT
-#define LORA_FIXED_HEADER_LEN_MODE_DEFAULT     (false)
+/** @brief Set this to 1 to enable frequency hopping
+ *
+ * If Frequency hopping spread spectrum (FHSS) is enabled a portion of each LoRa
+ * packet is transmitted on each hopping channel from a look up table of
+ * frequencies managed by the host microcontroller.
+*/
+#ifdef DOXYGEN
+#define CONFIG_LORA_FREQUENCY_HOPPING_DEFAULT
 #endif
 
-/** @brief Enable payload CRC, optional */
-#ifndef LORA_PAYLOAD_CRC_ON_DEFAULT
-#define LORA_PAYLOAD_CRC_ON_DEFAULT            (true)
+/** @brief Frequency hopping period
+ *
+ * Configure the hopping period, time which each transmission will dwell in any
```

Comment:
Could you add the unit here and in the Kconfig file? (unit is "symbols")

Reply:
Done